### ISSUE TITLE
Using ANSI in a build system only displays the right colors after the build has ended

### ISSUE THREAD

**Issue by SBRK** [aziz/SublimeANSI#8]
When the build is proceeding, I still see the "ESC" codes. Then when the build ends, the output takes the right colors.

here's what my sublime-build file looks like

```
{
    "shell_cmd": "...",

    /*  add target and syntax */
    "target": "ansi_color_build",
    "syntax": "Packages/ANSIescape/ANSI.tmLanguage"
}
```

**Comment by vmuriart** [comment]
Looks to be by [design](https://github.com/aziz/SublimeANSI/blob/master/ansi.py#L73)

**Comment by jakkubu** [comment]
Just solve it. Check it out: #9 There is small problem: the unformatted output is printed and after fraction of second it's replaced by properly formatted text.

**Comment by vmuriart** [comment]
I need to test it out a bit more. It looks like its also delayed the output;

**Comment by jakkubu** [comment]
You can always keep the old method and switch between these two according to new 'process_on_finsish' setting. After tests you can make on_data default since it's more universal or keep on_finish if you decide so.

**Comment by vmuriart** [comment]
Just tested out your latest commit (and realized what you meant by new setting), it works great! thanks

**Comment by matt1003** [comment]
A fix for this issue has now been released in ANSIescape version 0.1.3.
(note: this does not include the fix for the delayed output)

**Comment by matt1003** [comment]
The fix for delayed output has now been released in ANSIescape version 0.1.5.
